### DISCUSSION TITLE
Add Charybdis to Miryoku ZMK

### DISCUSSION THREAD

**Discussion by Qunos** [manna-harbour/miryoku#300] [Q&A] [answered]
Tried Void/EIGA wireless charybdis build from Bastard Keyboards ( https://www.youtube.com/watch?v=Mks7QDxFreY&t=122s ). The build tutorial and link for custom pcb for low power sensor is really awesome. Managed to build myself a wireless chary with no experience whatsoever.

Only problem i have is that cannot find a way to build a firmware for trackball on Miryoku.
Also, seems that mouse/trackball on ZMK is kind of limited.
Best i could find is this repo from Vloth (https://github.com/vloth/charybdis-zmk-module).

Wish for support with adding Charybdis in Miryoku.

**Comment by manna-harbour** [comment, marked as answer]
I've added support for modules and the pointers-move-scroll branch to Miryoku ZMK. In a charybdis testing branch I've also added the keymap, mapping, and outboard. After testing, that commit can be added to the main branch.

The keyboard definition includes default scroll and snipe layers, so they will need to be overridden. Ideally, they would be disabled in @vloth's module at the boards level and defined at the config level there instead (since they depend on the keymap), then set here if desired.

Select branch `charybdis-test` and build with board `nice_nano_v2`, shield `charybdis_left,charybdis_right`, and custom_config `#define MIRYOKU_KLUDGE_MOUSEKEYSPR`. Here's a sample build run: https://github.com/manna-harbour/miryoku_zmk/actions/runs/8307623682

**Comment by Qunos** [comment]
works perfectly. Awesome !!!

**Reply by manna-harbour** [reply]
Great! I'll add it to the main branch.

Do you have issues with the scroll and snipe mode activation on layer change? It looks like they would activate on layers 6 and 7 https://github.com/vloth/charybdis-zmk-module/blob/main/boards/shields/charybdis/charybdis_right.overlay#L61-L62, which would be Media and Num https://github.com/manna-harbour/miryoku_zmk/blob/master/miryoku/miryoku_babel/miryoku_layer_list.h#L16-L17.

**Reply by Qunos** [reply]
Did not notice any problems. Will test some more after i get new Nice Nano v2 and rebuild the keyboard. For some reason neither with Vloth, nor with Miryoku, my keyboard does not want to work over BT. It recognizes it, connects to it, but it does not work.
When i move the nice nanos from Chary over to Skeletyl, they work fine over BT.
When i connect to it via USB, keyboard works.
